crates/cxx-qt-lib/src/qml/qqmlapplicationengine.rs

```diff
@@ -101,4 +119,17 @@ impl QQmlApplicationEngine {
     pub fn new() -> cxx::UniquePtr<Self> {
         ffi::qqmlapplicationengine_new()
     }
+
+    /// Sets the provider to use for images requested via the image: url scheme, with host proveri_id. The QQmlEngine takes ownership of provider.
+    pub unsafe fn add_image_provider(self: Pin<&mut QQmlApplicationEngine>, provider_id: &QString, provider: QQmlImageProviderBasePointer) {
+        let ptr = match provider {
```

Comment:
this is another case where having up/down casting would help out #562 (something we hope for this cycle). As then the method could just support the QQmlImageProviderBase* and in Rust you do upcast_ptr() when calling the method.